### ISSUE TITLE
cmd/tailscaled: support running & starting directly, without a "tailscale up" step

### ISSUE THREAD

**Issue by bradfitz** [tailscale/tailscale#1412]
For a number of use cases (especially container-like stuff from #504), it'd be better to just run tailscaled with some args/env/config file/whatever and have it start up, rather than requiring a `tailscale up` step.

Tracking bug for that.

(This isn't about removing `tailscale up`; just adding an optional way to not require it.)

/cc @danderson

**Comment by gerardo-junior** [comment]
+1, --auto-connect on deamon to auto start for raspberry too

**Comment by lorenzleutgeb** [comment]
I am a happy but confused user, and, after some digging, arrived at the same conclusion. FWIW, here's my thought process:

After reading the 1.6 release mail and the [docs on exit nodes](https://tailscale.com/kb/1103/exit-nodes), I felt the urge to implement downstream support for exit nodes in NixOS. My idea would be to add a switch like

```nix
{
  services.tailscale = {
   enable = true;
   advertiseExitNode = true; # This is new.
  };
}
```

ATM NixOS [uses your systemd service](https://github.com/NixOS/nixpkgs/commit/503caab7765f31d18bce7fa58b26d447bc413c17), so I was wondering how I can "tell" `tailscaled` to advertise an exit node. What then resolved my confusion was the [Tailscale on NixOS](https://tailscale.com/kb/1096/nixos-minecraft) article, making the problem painstakingly obvious by introducing a systemd oneshot.

Please make "running and starting directly" possible, also for the sake of a more declarative way of getting up and running.

**Comment by vincentmele** [comment]
This also seems to mean that the Ubuntu debian file, which installs [tailscaled.service](https://github.com/tailscale/tailscale/blob/main/cmd/tailscaled/tailscaled.service) which invokes tailscaled (rather than tailscale with oneshot), does not support --advertise-exit-node out of the box. 

I'd much prefer to add the flag either to a config file in /etc/ or directly to tailscaled.service.

**Comment by DentonGentry** [comment]
For the ephemeral container use cases discussed here, having the container be an exit node seems unlikely.

For cases of long-running systems which do want to be an exit node, running tailscale up with --advertise-exit-node once should be sufficient. tailscaled will use the same flags again after future reboots.

**Comment by bradfitz** [comment]
> For the ephemeral container use cases discussed here, having the container be an exit node seems unlikely.

Well, this bug isn't just about ephemeral containers. It's about any sort of declarative infrastructure (containers, NixOS) where you want to describe a state and have us do it, without intermediate imperative steps that the user has to coordinate, like getting a cmd/tailscale to run at just the right time, and only once.

**Comment by vincentmele** [comment]
> For cases of long-running systems which do want to be an exit node, running tailscale up with --advertise-exit-node once should be sufficient. tailscaled will use the same flags again after future reboots.

If that can be documented somewhere, particularly on the page announcing / discussing how to enable exit nodes, it would help greatly. The connection between changing the state (?) or config by running `tailscale up` on an already configured and running tailscaled isn't obvious.

And this original bug submission does not about appear to be specific to containers. It was that containers are one example. In many use cases, running tailscaled without depending on `tailscale up` first, but especially when flags need to be changed, would be a very useful feature and more in line with how other daemon applications function.

**Comment by courtland** [comment]
Also chiming in for this request. I have 200 Linux nodes on my tailnet where `tailcale up` is run by a `oneshot` systemd unit. The systemd unit hangs (intentionally) until the authkey is accepted. The unit file is created programmatically with an IaC tool. This has worked alright... the realtime operator CLI experience with `tailscale up` is great, but this approach is lacking for headless systems driven by automation.

Now I want to enable the new auto-updates feature, and I have to do this with yet another oneshot systemd unit that runs `tailscale set`, I guess.

**Comment by irbekrm** [comment]
Still experimental and likely to change, but we added a support for passing config to tailscaled via a config file (in which case no `tailscale up/set` is ran) in https://github.com/tailscale/tailscale/pull/10759

**Comment by mpstefan** [comment]
As of now, Containerboot [automates the tailscale up command](https://github.com/tailscale/tailscale/blob/main/cmd/containerboot/main.go#L346). This could be used as a reference for if we want to automate it in the same way elsewhere. As this is no longer relevant to containers or Kubernetes, I am removing those tags so we no longer track it as a part of our backlog.
